### PR TITLE
ESQL: Known issue for OOMs for queries with wide field/index patterns

### PR DESCRIPTION
**Pull request by alex-spies**

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

**Review by luigidellaquila — APPROVED** [review]

File: docs/reference/release-notes/8.15.0.asciidoc [hunk 1]
```diff
@@ -30,6 +30,14 @@ signed integer) may encounter errors (issue: {es-issue}111854[#111854])
 `xpack.security.authc.realms.*.files.role_mapping` configuration option. As a workaround, custom role mappings
 can be configured using the https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html[REST API] (issue: {es-issue}112503[#112503])
 
+* ES|QL queries can cause nodes to crash due to Out Of Memory errors under some circumstances when there are many indices matching the pattern,
+many conflicting field mappings in those indices, and many of those fields are kept in the request. Lower heap proportional to all of that
+makes it worse. (issue: {es-issue}111964[#111964], {es-issue}111358[#111358]).
+Either upgrade to 8.15.2 upon release, or else follow the recommendation in the manual to
+[disable ES|QL](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-kibana.html#esql-kibana-enable) queries.
+You can also reduce the likelihood of users triggering the issue by changing the default data view in Discover to a smaller set of indices
+and/or one with fewer mapping conflicts.
```
Comment (leemthompo) [inline docs/reference/release-notes/8.15.0.asciidoc:39]:
```suggestion
* ES|QL queries can lead to node crashes due to Out Of Memory errors when:
** Multiple indices match the query pattern
** These indices have many conflicting field mappings
** Many of those fields are included in the request
+
These issues deplete heap memory, increasing the likelihood of OOM errors. (issue: {es-issue}111964[#111964], {es-issue}111358[#111358]).
+
To work around this issue, you have three options:
** Upgrade to 8.15.2 upon release
** Follow the instructions to
https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-kibana.html#esql-kibana-enable)[disable ES|QL] queries.
** Change the default data view in Discover to a smaller set of indices and/or one with fewer mapping conflicts.
```

@alex-spies Trying to reword for clarity, also the workarounds are very Discover centric not sure if that's intentional.

Comment (tylerperk) [reply docs/reference/release-notes/8.15.0.asciidoc:39]: @leemthompo thanks for reviewing. What do you think about having "Downgrade to an earlier version" as one of the workarounds?

Comment (leemthompo) [reply docs/reference/release-notes/8.15.0.asciidoc:39]: @tylerperk that also sounds reasonable

Comment (alex-spies) [reply docs/reference/release-notes/8.15.0.asciidoc:39]: Thanks, that's great! Applied to https://github.com/elastic/elasticsearch/pull/112926

**Comment by alex-spies** [comment]
Closing in favor of https://github.com/elastic/elasticsearch/pull/112926, as this needs to go to main (and be backported) after all.

**Pull request closed without merging**